### PR TITLE
Adding a configurable frame limiter.

### PR DESCRIPTION
**Pull request by Rhuagh**

### CODE REVIEW

File: src/util/frame_limiter.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
+//! Frame limiter
+
+use std::thread::{sleep, yield_now};
+use std::time::{Duration, Instant};
+
+/// Frame rate limiting strategy
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum FrameRateLimitStrategy {
+    /// No limit, spin as fast as possible
```
Comment (Xaeroxe) [inline src/util/frame_limiter.rs]: Would like to add "WARNING: THIS MAY PRODUCE SYSTEM INSTABILITY DUE TO CPU STARVATION"

File: src/app.rs [hunk 2]
```diff
@@ -120,10 +119,9 @@ impl<'a, 'b> Application<'a, 'b> {
         self.timer.start();
         while self.states.is_running() {
             self.advance_frame();
+
             // We are done with this frame, we can just rest for now.
```
Comment (Xaeroxe) [inline src/app.rs]: Might want to update this comment since we're not necessarily resting here.

File: src/util/frame_limiter.rs (hunk 1 above)
Comment (Xaeroxe) [reply src/util/frame_limiter.rs]: Alternatively: If the strategy for this variant was to call `yield_now` it might not be as much of a concern.

**Comment by Rhuagh** [comment]
Fixed @Xaeroxe.

**Comment by Rhuagh** [comment]
If needed, we can even do separate impl Default for different platforms.

**Pull request merged**